src/main/java/io/confluent/connect/jdbc/JdbcSourceConnector.java

```diff
@@ -154,8 +154,11 @@ protected CachedConnectionProvider connectionProvider(int maxConnAttempts, long
                 + "the list of tables from the database yet"
         );
       } else if (currentTables.isEmpty()) {
-        taskConfigs = Collections.emptyList();
-        log.warn("No tasks will be run because no tables were found");
+        taskConfigs = new ArrayList<>(1);
```

Comment:
do we need a similar behavior in the currentTables == null code branch as well?

Reply:
No I don't think so. If it's null it means that at the time configureTasks was called, the query was still blocking (even after @ncliang 's change).  We have to expect the query to conclude eventually and a request to reconfigure tasks to be called.